### ISSUE TITLE
Different encoding behavior for PersistDbSpecific with persistent-postgresql and persistent-mysql

### ISSUE THREAD

**Issue by sol** [yesodweb/persistent#1110]
## What is the discrepancy?
`persistent-mysql` and `persistent-postgresql` encode `PersistDbSpecific` differently.

`persistent-mysql` uses `Plain`:

https://github.com/yesodweb/persistent/blob/31e25c41c648fbbf5cca7cb79665482d008931bf/persistent-mysql/Database/Persist/MySQL.hs#L239

However, `persistent-postgresql` uses `Escape` (effectively, treating it as a string-like value):

https://github.com/yesodweb/persistent/blob/31e25c41c648fbbf5cca7cb79665482d008931bf/persistent-postgresql/Database/Persist/Postgresql.hs#L554

## When was this introduced?

The behavior for PostgreSQL changed with 9287943c1dd4ca0c9822a715c057f4fb2ab6888e.

## Why is this a problem?

With the current behavior it is not possible to encode something like `point '(2.0,0)'` for PostgreSQL.

(I would also be very surprised if the example given in the [docs for `PersistDbSpecific`](https://hackage.haskell.org/package/persistent-2.10.5.2/docs/Database-Persist-Types.html#v:PersistDbSpecific) is still anywhere close working; but I haven't tried that specifically.)

## How to address this?

I'm not sure how to best address this.  Altering how `PersistDbSpecific` is encoded would be a breaking change that isn't detected by the type checker.  One solution might be to remove `PersistDbSpecific` and add two new constructors:

- `PersistDbSpecificEscape`, which uses `Escape` (current behavior of `persistent-postgresql`)
- `PersistDbSpecificPlain`, which uses `Plain` (current behavior of `persistent-mysql`)

**Comment by parsonsmatt** [comment]
#1123 covers this in a PR

In #1122, I outline [a proposal](https://github.com/yesodweb/persistent/pull/1122#discussion_r515127688) for fixing the issue over a few release cycles.

tl;dr: Deprecate `PersistDbSpecific` in favor of `PersistLiteral` and `PersistLiteralEscaped`, basically your suggestion @sol :)
